research-software-directory/RSD-as-a-service
# Add search and bulk import to admin orcid-users page

# Add search and bulk import to admin/orcid-users page

Changes proposed in this pull request:
* rsd admin can search for ORCID in the orcid users list. Partial orcid value is supported.
* rsd admin can add single ORCID to the list, **only if ORCID is not found**.
* the name and affiliation for each orcid is pulled from orcid api during the list load
* rsd admin can add list of ORCID. The validation is performed before import.
* bulk import approach is similair to mention bulk import. Except that validation requests to ORCID are performed per orcid item
* updated documentation for ocrid-users administration page with new animation

How to test:
* `make start` to build app
* login as rsd admin. If you use same .env as I do you can use `professor3`
* navigate to adminstration orcid-users page
* try to add one orcid item
* search for that orcid item 
* delete that orcid item
* try to add a list of orcid items provided below. Have patience :-). Try to use invalid or non existing items too. Check documentation page too :-)
```bash
0000-0001-6435-2978
0000-0002-3789-8577
0000-0001-8120-3712
0009-0000-7257-0836
0000-0001-6238-6154
0000-0001-7116-3585
0000-0003-2342-2954
0000-0001-7158-454X
0009-0000-1619-0994
0000-0002-6024-1776
0000-0002-6524-4474
0000-0001-8832-9123
0000-0002-7441-7787
0000-0001-5981-8745
0000-0002-7745-6298
0000-0003-0157-8907
0000-0001-9937-0877
0000-0003-0869-998X
0000-0001-9972-9288
0000-0002-5642-7084
0009-0001-1022-4863
0000-0003-2396-8543
0000-0002-0318-316X
0000-0003-0685-1802
0000-0001-6663-9921
0000-0003-4229-4417
0000-0003-0351-8342
0000-0002-7643-170X
0000-0001-9824-1715
0000-0002-3931-4264
0000-0002-0002-7620
0000-0003-3035-6107
0000-0003-4418-131X
0000-0003-4420-389X
0000-0001-6717-883X
0000-0003-1721-7778
0000-0002-7250-4246
0000-0001-8714-7184
0000-0003-2439-2750
0000-0001-9119-5656
0000-0001-9772-554X
0000-0002-5123-2501
0009-0004-9127-2024
0009-0009-6811-2264
0009-0006-4911-5695
0000-0003-0590-0094
0000-0002-4951-8738
0000-0001-7115-1007
0000-0002-9854-7367
0000-0002-5029-3548
```
## Example no items
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/33d89483-f1c7-4e9f-8d54-f4e69c57f443)

## Example import one ORCID
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/e15710bf-2589-4865-ab67-83d8416174e0)

## Example import ORCID list
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/30a6fa88-0b08-448b-b421-6111b66184b7)

## Example ORCID list
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/04d4a970-74ef-4195-ae1d-d6e26a143f03)


PR Checklist:
